### ISSUE TITLE
custom plugins broken after update from 24.2.1 to 24.5.1

### ISSUE THREAD

**Issue by vic0824** [ArcadeData/arcadedb#1684]
### ArcadeDB Version:

ArcadeDB Server v24.5.1 (build 68008482ef7b0c52919f8d27acc53b49e41c877f/1716545156043/main)

### OS and JDK Version:

Linux 4.18.0-553.5.1.el8_10.x86_64 - OpenJDK 64-Bit Server VM 11.0.22 (Corretto-11.0.22.7.1)

### Expected behavior

custom plugins with custom Undertow routes, continue to work.

### Actual behavior

all custom routes return 404 (not found). Setting all logging levels to FINE does not produce any message that explains why Undertow cannot find the routes. Only the fallback handler works

### Steps to reproduce
working on a test case

**Comment by vic0824** [comment]
The attached zip file contains:

- testCustomPlugin: test method
- OrchestratorPlugin: the class that implements the custom plugin
- TestHandler: the class that implements the test handler

[OrchestratorPlugin.zip](https://github.com/user-attachments/files/16529655/OrchestratorPlugin.zip)

**Comment by vic0824** [comment]
Update: also v24.4.1 does not work, so something happened between 24.2.1 and 24.4.1 which broke the custom http routes

**Comment by vic0824** [comment]
I have found the root cause: up to 24.2.1, in `ArcadeDBServer`, the `registerPlugins()` method was called before `httpServer.startService()`.
Starting from 24.4.1, the `registerPlugins()` method is called after `httpServer.startService()`. and therefore the custom routes are not registered. 
I have pulled the latest snapshot from main, moved the `registerPlugins()` call before the `httpServer.startService()`, and now the custom routes work as before.

**Comment by lvca** [comment]
You're right, we moved that to fix a bug with HA. I think each plugin should declare when it wants to be installed, so I've created a new enum and a new method `getInstallationPriority` with default `BEFORE_HTTP_ON`:

```java
public interface ServerPlugin {
  enum INSTALLATION_PRIORITY {BEFORE_HTTP_ON, AFTER_HTTP_ON, AFTER_DATABASES_OPEN}

  default INSTALLATION_PRIORITY getInstallationPriority() {
    return BEFORE_HTTP_ON;
  }
}
```
